runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java

```diff
@@ -413,4 +423,44 @@ private boolean slotContainsText(UIActionSlot slot, String text)
 		return slot.getAction().getName().toLowerCase().contains(text.toLowerCase());
 	}
 
+	private FocusListener buildFocusListener()
+	{
+		return new FocusListener()
+		{
+			@Override
+			public void focusGained(FocusEvent e)
+			{
+				// No action needed for focusGained
+			}
+
+			@Override
+			public void focusLost(FocusEvent e)
+			{
+				try
```

Comment:
This feels very hacky. I think it would make more sense to have one listener per field and make this just callback to a Consumer/Callable. That way it could work similar to the action listeners- `field.addFocusListener(buildListener(e ->onFieldTargetXPUpdated()))`.  Also if you only want to override one method you can use `FocusAdapter`.